lib/apt_update_nonblocking_hook.py

```diff
@@ -0,0 +1,37 @@
+#!/usr/bin/python3
+
+import datetime
+import logging
+
+from uaclient import apt
+from uaclient.apt_news import fetch_and_process_apt_news
+from uaclient.config import UAConfig
+from uaclient.daemon import setup_logging
+
+
+def apt_news(cfg: UAConfig):
+    last_update = apt.get_apt_cache_datetime()
+    one_day_ago = datetime.datetime.now(
+        datetime.timezone.utc
+    ) - datetime.timedelta(days=1)
+    if last_update is not None and last_update > one_day_ago:
+        return
+    fetch_and_process_apt_news(cfg)
+
+
+def main(cfg: UAConfig):
```

Comment:
Do we need this function ? I think we can perform this check directly on `__main__`

Reply:
You're right, that extra function isn't helping us here - I'll simplify it

Reply:
done!